pyomo/gdp/plugins/bigm_mixin.py

```diff
@@ -0,0 +1,228 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.gdp import GDP_Error
+from pyomo.common.collections import ComponentSet
+from pyomo.contrib.fbbt.fbbt import compute_bounds_on_expr
+from pyomo.core import Suffix
+
+def _convert_M_to_tuple(M, constraint, disjunct=None):
+    if not isinstance(M, (tuple, list)):
+        if M is None:
+            M = (None, None)
+        else:
+            try:
+                M = (-M, M)
+            except:
+                logger.error("Error converting scalar M-value %s "
+                             "to (-M,M).  Is %s not a numeric type?"
+                             % (M, type(M)))
+                raise
+    if len(M) != 2:
+        constraint_name = constraint.name
+        if disjunct is not None:
+            constraint_name += " relative to Disjunct %s" % disjunct.name
+        raise GDP_Error("Big-M %s for constraint %s is not of "
+                        "length two. "
+                        "Expected either a single value or "
+                        "tuple or list of length two for M."
+                        % (str(M), constraint.name))
+
+    return M
+
+def _get_bigm_suffix_list(block, stopping_block=None):
+    # Note that you can only specify suffixes on BlockData objects or
+    # ScalarBlocks. Though it is possible at this point to stick them
+    # on whatever components you want, we won't pick them up.
+    suffix_list = []
+
+    # go searching above block in the tree, stop when we hit stopping_block
+    # (This is so that we can search on each Disjunct once, but get any
+    # information between a constraint and its Disjunct while transforming
+    # the constraint).
+    while block is not None:
+        bigm = block.component('BigM')
+        if type(bigm) is Suffix:
```

Comment:
Is this the right moment to unify the notation between `M`, `m`, `bigm`, `bigM`, `big_M`, ... ?

Reply:
Are you referring to the name of the Suffix? Or just the variable naming within the code?

Reply:
I refer to the variable naming within the code to try to make it more uniform across it. Some functions are for `m` others for `bigM`

Reply:
Just to emphasize what I was meaning, the next two functions in this file alone are called `_warn_for_unused_bigM_args` `_get_bigm_arg_list` and `_process_M_value`

Reply:
Yeah, I see your point. I've used `M` in the public API, so we should stick with that mostly. I like the idea of not using `bigm` since that overload the name of the transformation. I switched the ones that are specifically referring to args and Suffixes to `bigM` and tried to leave everything else as `M`.